lib/validators/omnivore.js

```diff
@@ -38,7 +40,56 @@ module.exports = function validateOmnivore(opts, callback) {
       if (size > limits.max_filesize) {
         return callback(invalid('File is larger than ' + prettyBytes(limits.max_filesize) + '. Too big to process.'));
       }
-      callback();
+      //only look at files that passed above file size test
+      files.forEach(function(f) {
+        q.defer(validateGdal, f, filetype);
+      });
+      q.awaitAll(function(err) {
+        if (err) return callback(err);
+        callback();
+      });
     });
   });
+
+  function validateGdal(file, filetype, callback){
```

Comment:
There is not actually any code in this function that runs asynchronously, right? Because of this, the function will be clearer if you do not use a callback. Instead:

```js
function validateGdal(file, filetype);
```

It looks like the function would have three possible outcomes:
- returns nothing
- returns a string indicating that there was a validation error
- throws an error if some synchronous gdal step fails